backend/api/Controllers/BuildingController.cs

```diff
@@ -93,6 +93,10 @@ public IActionResult GetBuildings([FromBody]BuildingFilter filter)
         public IActionResult GetBuilding(int id)
         {
             var entity = _pimsService.Building.Get(id);
+            if(entity.Address?.Postal != null)
+            {
+                entity.Address.Postal = entity.Address.Postal.Substring(0,3) + " " + entity.Address.Postal.Substring(3,3);
```

Comment:
You can do this `entity.Address.Postal.Insert(3, " ")`, and since you are doing it multiple places, add an extension method to `StringExtensions.FormatAsPostal(...)`.

Reply:
Also since this is the response, move logic to the mapper instead.